packages/lib/getFeatures/wfs/match.ts

```diff
@@ -83,23 +85,31 @@ const sortMatches = (
     known.push(asString)
     return true
   })
-
-  return sortedFilteredMatches
 }
 
 /**
  * matches an input string to patterns
  */
 export const match = (
-  patterns: string[],
-  patternKeys: Record<string, string>,
+  patterns: string[] | undefined,
+  patternKeys: Record<string, string> | undefined,
   inputValue: string
-): string[][][] => {
-  const matches: string[][][] = []
+): KeyValueSetArray => {
+  if (!patterns) {
+    throw new Error(
+      '@polar/lib-get-features: Parameter "patterns" is missing on wfs configuration for pattern-based search.'
+    )
+  }
+  if (!patternKeys) {
+    throw new Error(
+      '@polar/lib-get-features: Parameter "patternKeys" is missing on wfs configuration for pattern-based search.'
+    )
+  }
+  const matches: KeyValueSetArray = []
   const uninterpretedCharacters: number[] = []
   patterns.forEach((pattern) => {
     const patternBlocks = getBlocks(pattern)
-    const patternMapping: string[][] = []
+    const patternMapping: Array<[string, string]> = []
```

Comment:
`KeyValueSetArray[number]` is an "access" to the type, returning the type that's in the array.

That is, on a type `type stringray = string[]`, `stringray[number]` equals `string`, since accessing contents of `stringray` with a number yields a `string` (or undefined, but we don't assume this in TS).

Reply:
That sounds - reasonable :D  
Will implement it like that!